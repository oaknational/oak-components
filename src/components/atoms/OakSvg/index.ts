export { OakSvg } from "./OakSvg";
export type { OakSvgProps, OakSvgNames } from "./OakSvg";
