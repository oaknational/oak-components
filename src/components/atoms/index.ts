export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
