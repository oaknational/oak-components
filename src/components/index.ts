export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
