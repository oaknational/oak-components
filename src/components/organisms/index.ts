export * from "./pupil";
export * from "./teacher";
export * from "./OakSubjectIcon";
export * from "./OakPrimaryNavItem";
export * from "./OakPrimaryNav";
export * from "./OakCookieBanner";
export * from "./OakCookieSettingsModal";
export * from "./OakCookieConsentProvider";
export * from "./OakCookieConsent";
