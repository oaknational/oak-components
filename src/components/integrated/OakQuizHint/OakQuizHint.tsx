import React, { MouseEventHandler, ReactNode, useState } from "react";

import { OakTertiaryButton, OakTooltip } from "@/components/ui";

export type OakQuizHintProps = {
  /**
   * Some content to give as a hint to answer a question
   */
  hint: ReactNode;
};

/**
 * Presents a button which will show a hint when clicked
 */
export const OakQuizHint = ({ hint }: OakQuizHintProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const handleClick: MouseEventHandler = () => {
    setIsOpen(!isOpen);
  };

  return (
    <OakTooltip tooltip={hint} isOpen={isOpen}>
      <OakTertiaryButton
        iconName={isOpen ? "lightbulb-yellow" : "lightbulb"}
        onClick={handleClick}
        iconBackground={isOpen ? "bg-icon" : "bg-decorative5-main"}
      >
        {isOpen ? "Close hint" : "Need a hint?"}
      </OakTertiaryButton>
    </OakTooltip>
  );
};
