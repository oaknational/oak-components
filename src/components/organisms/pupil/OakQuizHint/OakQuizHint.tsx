import React, { MouseEventHandler, ReactNode, useState } from "react";

import { OakTooltip } from "@/components/molecules";
import { OakHintButton } from "@/components/organisms/pupil/OakHintButton";

export type OakQuizHintProps = {
  /**
   * Some content to give as a hint to answer a question
   */
  hint: ReactNode;
};

/**
 * Presents a button which will show a hint when clicked
 */
export const OakQuizHint = ({ hint }: OakQuizHintProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const handleClick: MouseEventHandler = () => {
    setIsOpen(!isOpen);
  };

  return (
    <OakTooltip tooltip={hint} isOpen={isOpen} id="hint-tooltip">
      <OakHintButton
        isOpen={isOpen}
        onClick={handleClick}
        buttonProps={{ "aria-describedby": "hint-tooltip" }}
      />
    </OakTooltip>
  );
};
