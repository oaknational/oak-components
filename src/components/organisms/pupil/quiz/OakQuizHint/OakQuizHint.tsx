import React, { MouseEventHandler, ReactNode, useState } from "react";

import { OakTooltip } from "@/components/molecules";
import { OakHintButton } from "@/components/organisms/pupil/quiz/OakHintButton";
import { OakBox } from "@/components/atoms";

export type OakQuizHintProps = {
  /**
   * Some content to give as a hint to answer a question
   */
  hint: ReactNode;
  id: string;
  hintToggled?: (props: { isOpen: boolean }) => void;
};

/**
 * Presents a button which will show a hint when clicked
 */
export const OakQuizHint = ({ hint, id, hintToggled }: OakQuizHintProps) => {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const handleClick: MouseEventHandler = () => {
    setIsOpen(!isOpen);
    if (hintToggled) {
      hintToggled({ isOpen: !isOpen });
    }
  };

  return (
    <>
      <OakBox $display="none" id={id}>
        {hint}
      </OakBox>
      <OakTooltip tooltip={hint} isOpen={isOpen} tooltipPosition="top-left">
        <OakHintButton
          isOpen={isOpen}
          onClick={handleClick}
          buttonProps={{ "aria-describedby": id }}
        />
      </OakTooltip>
    </>
  );
};
