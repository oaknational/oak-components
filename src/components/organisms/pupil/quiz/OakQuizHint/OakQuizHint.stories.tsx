import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakQuizHint } from "./OakQuizHint";

import { OakBox } from "@/components/atoms";

const meta: Meta<typeof OakQuizHint> = {
  component: OakQuizHint,
  tags: ["autodocs"],

  argTypes: {
    hint: {
      control: {
        type: "text",
      },
    },
  },
  decorators: [
    (Story) => (
      <OakBox $mt="space-between-xxl">
        <Story />
      </OakBox>
    ),
  ],
  args: {
    hint: "A football and an orange are both sphere shaped, like Earth.",
    id: "quiz-hint",
  },
};
export default meta;

type Story = StoryObj<typeof OakQuizHint>;

export const Default: Story = {
  render: (args) => <OakQuizHint {...args} />,
};
