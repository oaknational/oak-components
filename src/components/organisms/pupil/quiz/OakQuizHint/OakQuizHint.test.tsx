import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";

import { OakQuizHint } from "./OakQuizHint";

import { OakThemeProvider } from "@/components/atoms";
import { oakDefaultTheme } from "@/styles";
import { fireEvent } from "@testing-library/react";
import renderWithTheme from "@/test-helpers/renderWithTheme";
import {
  installMockIntersectionObserver,
  installMockResizeObserver,
} from "@/test-helpers";

installMockResizeObserver();
installMockIntersectionObserver();

describe(OakQuizHint, () => {
  it("matches snapshot", () => {
    const tree = create(
      <OakThemeProvider theme={oakDefaultTheme}>
        <OakQuizHint
          hint="The answer is right in front of your eyes"
          id="quiz-hint"
        />
        ,
      </OakThemeProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it("calls hintToggled when hint button is clicked", () => {
    const hintToggled = jest.fn();
    const { getByRole } = renderWithTheme(
      <OakThemeProvider theme={oakDefaultTheme}>
        <OakQuizHint
          hint="The answer is right in front of your eyes"
          id="quiz-hint"
          hintToggled={hintToggled}
        />
        ,
      </OakThemeProvider>,
    );

    fireEvent.click(getByRole("button"));

    expect(hintToggled).toHaveBeenCalledTimes(1);
    expect(hintToggled).toHaveBeenCalledWith({ isOpen: true });
  });
});
