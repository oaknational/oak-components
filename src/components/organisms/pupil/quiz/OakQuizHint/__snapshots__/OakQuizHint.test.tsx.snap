// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakQuizHint matches snapshot 1`] = `
.c0 {
  display: none;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c1 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c7 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c9 {
  position: relative;
  width: 2.5rem;
  min-width: 2.5rem;
  height: 2.5rem;
  color: #222222;
  background: #ffe555;
  border-radius: 6.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c11 {
  position: absolute;
  top: 0rem;
  width: 100%;
  height: 100%;
  border-radius: 6.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c12 {
  position: relative;
  width: 1.5rem;
  min-width: 1.5rem;
  height: 1.5rem;
  min-height: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c13 {
  object-fit: contain;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.75rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c5 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: --var(google-font),Lexend,sans-serif;
  color: #222222;
}

.c5:disabled {
  pointer-events: none;
  cursor: default;
}

.c6 {
  display: inline-block;
  position: relative;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #222222;
}

.c6:active {
  color: #222222;
}

.c6:disabled {
  color: #808080;
}

.c3 > :first-child:focus-visible .shadow {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c3 > :first-child:hover .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

.c3 > :first-child:active .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%), 0.25rem 0.25rem 0 rgba(87,87,87,100%);
}

.c3 > :first-child:disabled .icon-container {
  background: #808080;
}

.c3 > :first-child:hover .icon-container {
  background: #ffe555;
}

.c3 > :first-child:active .icon-container {
  background: #ffe555;
}

.c4:hover .shadow {
  box-shadow: none !important;
}

.c4:active .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%), 0.25rem 0.25rem 0 rgba(87,87,87,100%) !important;
}

<div>
  <div
    class="c0"
    id="quiz-hint"
  >
    The answer is right in front of your eyes
  </div>
  <div
    style="display: contents;"
  >
    <div
      class="c1 c2 c3 c4"
    >
      <button
        aria-describedby="quiz-hint"
        class="c5 c6"
        data-rac="true"
      >
        <div
          class="c7 c8"
        >
          <div
            class="c9 c10 icon-container"
          >
            <div
              class="c11 shadow"
            />
            <div
              class="c12"
              data-icon-for="button"
            >
              <img
                alt=""
                class="c13"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1705078631/icons/zldisxmbff36z68rwcef.svg"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
          </div>
          <span
            class="c14"
          >
            Need a hint?
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
